docs/wcs/index.rst

```diff
@@ -28,89 +42,139 @@ It performs three separate classes of WCS transformations:
+A&A 446, 747 <https://doi.org/10.1051/0004-6361:20053818>`_).
 
-    4. Use one of the following transformation methods:
+However, there’s a difference in what is considered to be the first pixel. The
+``High Level API`` follows the Python and C convention that the first pixel is
+the 0-th one, i.e. the first pixel spans pixel values -0.5 to + 0.5. The
+``Low Level API`` takes an additional ``origin`` argument with values of 0 or 1
+indicating whether the input arrays are 0- or 1-based.
+The Low-level interface assumes Cartesian order (x, y) of the input coordinates,
+however the Common Interface for World Coordinate System accepts both conventions.
+The order of the pixel coordinates ((x, y) vs (row, column)) in the Common API
+depends on the method or property used, and this can normally be determined from
+the property or method name. Properties and methods containing “pixel” assume (x, y)
+ordering, while properties and methods containing “array” assume (row, column) ordering.
 
-       - From pixels to world coordinates:
+A Simple Example
+================
 
-         - `~astropy.wcs.wcs.WCS.all_pix2world`: Perform all three
-           transformations in series (core WCS, SIP and table lookup
-           distortions) from pixel to world coordinates.  Use this one
-           if you're not sure which to use.
+The following example shows how to load an imaging WCS from an extension HDU.
+The steps can be generalized to the case of a FITS file of any dimensions::
 
-         - `~astropy.wcs.wcs.WCS.wcs_pix2world`: Perform just the core
+    >>> from astropy.wcs import WCS
+    >>> from astropy.io import fits
+    >>> from astropy.utils.data import get_pkg_data_filename
+    >>> fn = get_pkg_data_filename('data/j94f05bgq_flt.fits', package='astropy.wcs.tests')
+    >>> hdul = fits.open(fn)
+    >>> hdul.info()  #doctest: +SKIP
+    Filename: /root/project/.tox/py36-test/lib/python3.6/site-packages/astropy/wcs/tests/data/j94f05bgq_flt.fits
+    No.    Name      Ver    Type      Cards   Dimensions   Format
+      0  PRIMARY       1 PrimaryHDU     251   ()
+      1  SCI           1 ImageHDU       184   (1, 1)   float32
+      2  ERR           1 ImageHDU        69   ()
+      3  DQ            1 ImageHDU        69   ()
+      4  SCI           2 ImageHDU       184   (1, 1)   float32
+      5  ERR           2 ImageHDU        69   ()
+      6  DQ            2 ImageHDU        69   ()
+    >>> data = hdul[1].data
+    >>> hdr = hdul[1].header
+    >>> wcs = WCS(hdr)
+    >>> wcs  #doctest: +SKIP
+    WCS Keywords
+
+    Number of WCS axes: 2
+    CTYPE : 'RA---TAN-SIP'  'DEC--TAN-SIP'
+    CRVAL : 5.63056810618  -72.05457184278998
+    CRPIX : 2048.0  1024.0
+    CD1_1 CD1_2  : 1.29056256197165e-05  5.95309123310338e-06
+    CD2_1 CD2_2  : 5.0220581265601e-06  -1.2644774105568e-05
+    NAXIS : 1  1
+
+It is a good habit to check how many extensions are there, in case the FITS file is
+of the multi-extension type.
+
```

Comment:
I think the `Simlpe example` section should use the high level interface. All we need is a call to `wcs.pixel_to_world` and a call to `wcs.world_to_pixel`, and printing the results.

The current example should go under the `Low Level Interface` section.

Reply:
Yup, have added one example for `wcs. pixel_to_world` and another one for `wcs.world_to_pixel` in the "A Simple Example" section. (Maybe I should rename this to "Simple Examples" next time?)